src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -667,11 +667,38 @@ private SourceFeature addSourceQualifiers( TaxonHelper taxonHelper, SampleInfo s
 			select_sourcequalifiers_pstmt = connection.prepareStatement(select_sourcefeature_Query);
 			select_sourcequalifiers_pstmt.setString(1, sampleInfo.getSampleId());
 			select_sourcequalifers_rs = select_sourcequalifiers_pstmt.executeQuery();
+			String latitude = null;
+			String longitude = null;
+			String country = null;
 			while (select_sourcequalifers_rs.next()) {
 				String tag = select_sourcequalifers_rs.getString(1);
 				String value = select_sourcequalifers_rs.getString(2);
+				if (sourceUtils.isCovidTaxId(sourceFeature.getTaxId()) && tag != null) {
+					if (tag.toLowerCase().contains("latitude")) {
+						latitude = value;
+						continue;
+					}
+					if (tag.toLowerCase().contains("longitude")) {
+						longitude = value;
+						continue;
+					}
+					if (tag.trim().equalsIgnoreCase("geographic location (country and/or sea)")) {
+						country = country == null ? value : value +":"+country;
+						continue;
+					}
+					if (tag.trim().equalsIgnoreCase("geographic location (region and locality)")) {
+						country = country == null ? value : country +":"+value;
+						continue;
+					}
```

Comment:
Are we sure we get this in the right order or would be safer to store the 'geographic location (country and/or sea' and 'geographic location (region and locality)' in two different variables 'country' and 'region' and then later construct the final value:

```
if(country != null) {
    String countryQualifierValue = country;
    if (region != null) {
       countryQualifierValue += ":" + region;
   }
   sourceUtils.addSourceQualifier(Qualifier.COUNTRY_QUALIFIER_NAME, country, sourceFeature);
}
```

i.e. do a similar thing you are doing with latitude and longitude.